### DISCUSSION TITLE
Isolation level is always null, if queried from the connection

### DISCUSSION THREAD

**Discussion by domkun** [eclipse-rdf4j/rdf4j#4798] [Q&A] [answered]
Hi there,

I had a problem with concurrent transactions, so I wanted to log the isolation level of the current connection using `RepositoryConnection#getIsolationLevel` method. However it always returns `null`. The underlying `sailConnection` property of the `SailRepositoryConnection` shows the set isolation level. Is this intended?

I am a bit confused about this. Here is a short reproducer:

```java
import org.eclipse.rdf4j.common.transaction.IsolationLevels;
import org.eclipse.rdf4j.repository.Repository;
import org.eclipse.rdf4j.repository.RepositoryConnection;
import org.eclipse.rdf4j.repository.sail.SailRepository;
import org.eclipse.rdf4j.sail.memory.MemoryStore;

class Scratch {
    public static void main(String[] args) {
        Repository sailRepository = new SailRepository(new MemoryStore());

        try (RepositoryConnection connection = sailRepository.getConnection()) {
            connection.begin(IsolationLevels.SERIALIZABLE);
            System.out.println(connection.getIsolationLevel());
            connection.commit();
        }
    }
}
```
Edit: Screenshot of the debugger window attached

![image](https://github.com/eclipse-rdf4j/rdf4j/assets/1139208/6f18add5-ccbb-405e-9155-3a215623f6ec)

**Comment by hmottestad** [comment, marked as answer]
Seems like a bug.

**Reply by domkun** [reply]
Thanks for the quick reply and creating a issue. I was not sure if the two properties should be in sync or have a different meaning.

**Reply by hmottestad** [reply]
They should be, yes. I assume that it stopped working when we started using TransactionSetting.
